### PR TITLE
Refactor OVM Get Contracts

### PR DESCRIPTION
**Pull request by MSilb7**
Brief comments on the purpose of your changes:

Dune v1 - Refactored the Optimism contract inserts so it runs faster and requires less updating. I believe the cron from this was paused, can we re-run it?

Once this gets through, I can create an update script to overwrite some contract with missing created times that the prior version overwrote. An alternative is to delete and re-write this table.

*For Dune Engine V2*
I've checked that:

* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`

When you are ready for a review, tag duneanalytics/data-experience. We will re-open your forked pull request as an internal pull request. Then your spells will run in dbt and the logs will be avaiable in Github Actions DBT Slim CI. This job will only run the models and tests changed by your PR compared to the production project.

### CODE REVIEW

**Comment by jeff-dude** [comment]
@MSilb7 i've rebuilt all objects as intended here & paused both cron jobs. how do you want me to proceed prior to resuming cron jobs? update statement as mentioned or drop all delete and restart (if so, which table(s))?

**Comment by MSilb7** [comment]
> @MSilb7 i've rebuilt all objects as intended here & paused both cron jobs. how do you want me to proceed prior to resuming cron jobs? update statement as mentioned or drop all delete and restart (if so, which table(s))?

Hi awesome! Dropping works, I actually want to add one additional field if we're doing that. So I'll make that change, then get on the order of what to update.

**Comment by jeff-dude** [comment]
> > @MSilb7 i've rebuilt all objects as intended here & paused both cron jobs. how do you want me to proceed prior to resuming cron jobs? update statement as mentioned or drop all delete and restart (if so, which table(s))?
> 
> Hi awesome! Dropping works, I actually want to add one additional field if we're doing that. So I'll make that change, then get on the order of what to update.

sounds good, just tag me here when ready

**Comment by MSilb7** [comment]
> > > @MSilb7 i've rebuilt all objects as intended here & paused both cron jobs. how do you want me to proceed prior to resuming cron jobs? update statement as mentioned or drop all delete and restart (if so, which table(s))?
> > 
> > 
> > Hi awesome! Dropping works, I actually want to add one additional field if we're doing that. So I'll make that change, then get on the order of what to update.
> 
> sounds good, just tag me here when ready

Made an update to add in the creation tx hash as a field.

Order:
1. Drop `ovm2.get_contracts`
2. Re-Create `ovm2.get_contracts`
3. Run insert - should take ~2 mins
```SELECT ovm2.insert_get_contracts('01-01-2021'::timestamptz,NOW())
WHERE NOT EXISTS (
SELECT *
FROM ovm2.get_contracts
);
```
4. Re-enable crons in `insert_get_contracts.sql` and `insert_updated_contract_info.sql`

File: optimism2/ovm2/get_contracts/insert_get_contracts.sql [hunk 1]
```diff
@@ -174,7 +174,7 @@ GROUP BY 1,2,3,4,5,6
                   fifth_level -- check for contract factories 4 layers down --as of 3/12, this ran through all contracts
               ) con
               LEFT JOIN ovm2."contract_creator_address_list" cc ON con.creator_address = cc.creator_address
-              LEFT JOIN ovm2."contract_creator_address_list" ccf ON con.creator_address = ccf.contract_creator_if_factory
+              LEFT JOIN ovm2."contract_creator_address_list" ccf ON con.creator_address = ccf.creator_address
```
Comment (MSilb7) [inline optimism2/ovm2/get_contracts/insert_get_contracts.sql:177]: Ah had a typo here, will fix. Shouldn't create any major impact for now, but I'll check. Can be fixed with an update if needed.

**Pull request merged**